service/collector/aws.go

```diff
@@ -0,0 +1,85 @@
+package collector
+
+import (
+	"github.com/giantswarm/microerror"
+	v1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+
+	"github.com/giantswarm/aws-operator/client/aws"
+	awsservice "github.com/giantswarm/aws-operator/service/aws"
+	"github.com/giantswarm/aws-operator/service/controller/v15/credential"
+)
+
+// getARNs list all unique aws IAM ARN from credential secret.
+func (c Collector) getARNs() ([]string, error) {
+	var ARNs []string
+
+	// List AWSConfigs.
+	awsConfigClient := c.g8sClient.ProviderV1alpha1().AWSConfigs("")
+	awsConfigs, err := awsConfigClient.List(v1.ListOptions{})
+	if err != nil {
+		return nil, microerror.Mask(err)
+	}
+
+	// Get unique ARNs.
+	arnsMap := make(map[string]bool)
+	for _, awsConfig := range awsConfigs.Items {
+		arn, err := credential.GetARN(c.k8sClient, &awsConfig)
+		if err != nil {
+			return nil, microerror.Mask(err)
+		}
+		arnsMap[arn] = true
+	}
+
+	for arn, _ := range arnsMap {
+		ARNs = append(ARNs, arn)
+	}
+
+	return ARNs, nil
+}
+
+// getAWSClients return a list of aws clients for every guest cluster account plus
+// the host cluster account.
+func (c Collector) getAWSClients() ([]aws.Clients, error) {
+	var clients []aws.Clients
+
+	ARNs, err := c.getARNs()
```

Comment:
I'm fuzzy on idiomatic golang variable naming, but should `ARNs` be `arns`?

Reply:
Probably yes